hazelcast/src/main/java/com/hazelcast/config/AbstractBasicSymmetricEncryptionConfig.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+
+/**
+ * Base class for symmetric encryption configuration classes.
+ *
+ * @param <T> the type of the configuration class
+ */
+public abstract class AbstractBasicSymmetricEncryptionConfig<T extends AbstractBasicSymmetricEncryptionConfig> {
```

Comment:
Minor: Can we call just `AbstractSymmetricEncryptionConfig`?

Reply:
Will do.